jsapp/js/account/organization/OrganizationSettingsField.tsx

```diff
@@ -0,0 +1,36 @@
+import TextBox from 'jsapp/js/components/common/textBox';
+import organizationSettingsStyles from 'js/account/organization/organizationSettingsRoute.module.scss';
+
+interface Props {
+  label: string;
+  value: string;
+  isDisabled?: boolean;
+  /** If `onChange` is not provided, we make the field disabled for safety. */
+  onChange?: (newValue: string) => void;
+  /**
+   * Function that ensures that field value is valid. If invalid will cause
+   * an error to be displayed.
+   */
+  validateValue?: (currentValue: string) => string | boolean | string[] | undefined;
+}
+
+/**
+ * A `TextBox` wrapper componet for `OrganizationSettingsRoute` that makes code
+ * a bit more DRY.
+ */
+export default function OrganizationSettingsField(
+  {label, value, isDisabled, onChange, validateValue}: Props
+) {
+  return (
+    <div className={organizationSettingsStyles.field}>
+      <TextBox
+        label={label}
+        value={value}
+        required
+        onChange={onChange}
+        disabled={!onChange || isDisabled}
+        errors={validateValue ? validateValue(value) : undefined}
```

Comment:
🤔 

the logic here seems to be reversed, or the naming may be confusing. 

`validateValue` seems like it should return `true` if the field is valid.

Maybe we could just have the same `errors` prop as boolean being set in the parent and passed through?

Reply:
I've used to `validate*` returning list of errors because that's what form libs (e.g. [formik](https://formik.org/docs/guides/validation)) I worked with before did. This TextBox also displays error so that seems right to me. Do you have examples or suggestions for a better naming?

Reply:
Changed :+1: